src/search/hnsw_indexer.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include "search/indexer.h"
+#include "search/search_encoding.h"
+#include "search/value.h"
+#include "storage/storage.h"
+
+namespace redis {
+
+class HnswIndex;
+
+struct Node {
```

Comment:
`Node` seems to generic. I think we can rename it to something like `HnswNode`.

Reply:
Done :)